src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -31,113 +29,214 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
-                        break; //Do not necessary check others elements
-                    }
+        if (isLeftStatement(u) || isRightStatement(u)) {
+            getStatementAssociatedWithUnit(u).forEach(valueBox -> {
+                Statement stmt = assignStatement(u);
+
+                if (valueBox.getValue() instanceof Local) {
+                    Local local = (Local) valueBox.getValue();
+                    res.add(new DataFlowAbstraction(local, stmt));
+                } else if (valueBox.getValue() instanceof StaticFieldRef) {
+                    StaticFieldRef staticFieldRef = (StaticFieldRef) valueBox.getValue();
+                    res.add(new DataFlowAbstraction(staticFieldRef, stmt));
+                } else if (valueBox.getValue() instanceof JArrayRef) {
+                    JArrayRef jArrayRef = (JArrayRef) valueBox.getValue();
+                    res.add(new DataFlowAbstraction(jArrayRef, stmt));
                 }
             });
         }
         return res;
     }
 
-
+    /*
+     * Here we divide the content of "in" into two lists
+     * left and right based on the type of your statement.
+     *
+     * To check if there is a possible conflict we compare
+     * the unit of right with the list of left
+     * or the unit of left with the list of Right
+     */
     @Override
-    protected void detectConflict(FlowSet<DataFlowAbstraction> in, Unit u){
-        if (!(isSinkStatement(u) || isSourceStatement(u))){
+    protected void detectConflict(FlowSet<DataFlowAbstraction> in, Unit u) {
+        if (!(isRightStatement(u) || isLeftStatement(u))) {
             return;
         }
+
         List<DataFlowAbstraction> left = new ArrayList<>();
         List<DataFlowAbstraction> right = new ArrayList<>();
 
-        u.getUseAndDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
-            String localName = getLocalName((Local) v.getValue());
-            for (DataFlowAbstraction filterIn: in){
-                String inName = getLocalName(filterIn.getLocal());
+        getStatementAssociatedWithUnit(u).forEach(valueBox -> {
+            String localName = getLocalValueBoxName(valueBox);
+
+            for (DataFlowAbstraction filterIn : in) {
+                String inName = getLocalDataFlowAbstraction(filterIn);
 
-                if (filterIn.getStmt().getType().equals(Statement.Type.SOURCE) && inName.equals(localName)){
+                if (filterIn.getStmt().getType().equals(Statement.Type.SOURCE) && inName.equals(localName)) {
                     left.add(filterIn);
-                }else if (filterIn.getStmt().getType().equals(Statement.Type.SINK) && inName.equals(localName)){
+                } else if (filterIn.getStmt().getType().equals(Statement.Type.SINK) && inName.equals(localName)) {
                     right.add(filterIn);
                 }
             }
         });
 
-        if(isSinkStatement(u)) {
+        if (isRightStatement(u)) {
```

Comment:
@barbosamaatheus in retrospective, would the code be a more clear implementation?

if (isRightStatement(u)) {
    pontentialConflictingAssignments = in.filter(abstraction ->  abstraction.containsLeftStatement())
} else if (isLeftStatement(u)) {
    pontentialConflictingAssignments = in.filter(abstraction ->  abstraction.containsRightStatement())
}
getStatementAssociatedWithUnit(u).forEach(valueBox -> {
             String localName = getLocalName((Local) v.getValue());	
             if isVariableNameInAssignments( localName, pontentialConflictingAssignments)	 {
                   createConflit(...)
             }
}

Reply:
Done!

Reply:
@barbosamaatheus dá pra ser bem mais simples. este código

f (isRightStatement(u)) {
pontentialConflictingAssignments = in.filter(abstraction -> abstraction.containsLeftStatement())
} else if (isLeftStatement(u)) {
pontentialConflictingAssignments = in.filter(abstraction -> abstraction.containsRightStatement())
}
getStatementAssociatedWithUnit(u).forEach(valueBox -> {
String localName = getLocalName((Local) v.getValue());
if isVariableNameInAssignments( localName, pontentialConflictingAssignments) {
createConflit(...)
}
}

apareceria logo após 

if (!(isRightStatement(u) || isLeftStatement(u))) {
             return;
         }

não precisaria nem criar nem popular left e right